<app-header></app-header>
<main *ngIf="result$ | async as result; else noResult" class="app-main">
    <header class="app-title">
        Measurement result from {{ result.measurementDate | date }}
    </header>
    <div class="app-wrapper">
        <app-dl
            [items]="{
                'Server name': result.measurementServerName,
                'IP Address': result.ipAddress,
                Provider: result.providerName
            }"
        ></app-dl>
        <app-spacer></app-spacer>
        <app-dl
            [items]="{
                Download: getSpeedInMbps(result.downloadKbit),
                Upload: getSpeedInMbps(result.uploadKbit),
                Ping: getPingInMs(result.ping)
            }"
        ></app-dl>
    </div>
    <footer class="app-footer">
        <a routerLink="/" class="app-btn">Start a new test</a>
    </footer>
</main>
<app-footer></app-footer>

<ng-template #noResult>
    <main *ngIf="error$ | async as error" class="app-main">
        There was an error while processing the result. {{ error.message }}
    </main>
    <main *ngIf="(error$ | async) === null" class="app-main">
        Fetching the result...
    </main>
</ng-template>
